pmd-java/src/main/resources/category/java/performance.xml

```diff
@@ -206,7 +206,15 @@ adverse impacts on performance.
             <property name="xpath">
                 <value>
 <![CDATA[
-//PrimitiveType[@Image = 'short'][name(../..) != 'CastExpression']
+//FieldDeclaration/Type/PrimitiveType[@Image = 'short']
+|
+//ClassOrInterfaceBodyDeclaration[not(Annotation/MarkerAnnotation/Name[@Image='Override'])]
```

Comment:
consider using `typeof()` to also support (the unlikely) `@java.lang.Override` syntax